nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/ScanKudu.java

```diff
@@ -0,0 +1,376 @@
+@Tags({"kudu", "scan", "fetch", "get"})
+@CapabilityDescription("Scans rows from a Kudu table with an optional list of predicates")
+@WritesAttributes({
+        @WritesAttribute(attribute = "kudu.table", description = "The name of the Kudu table that the row was fetched from"),
+        @WritesAttribute(attribute = "mime.type", description = "Set to application/json when using a Destination of flowfile-content, not set or modified otherwise"),
+        @WritesAttribute(attribute = "kudu.rows.count", description = "Number of rows in the content of given flow file"),
+        @WritesAttribute(attribute = "scankudu.results.found", description = "Indicates whether at least one row has been found in given Kudu table with provided predicates. "
+                + "Could be null (not present) if transfered to FAILURE")})
+public class ScanKudu extends AbstractKuduProcessor {
+
+    static final Pattern PREDICATES_PATTERN = Pattern.compile("\\w+((<=|>=|[=<>])(\\w|-)+)?(?:,\\w+((<=|>=|[=<>])(\\w|-)+)?)*");
+    static final Pattern COLUMNS_PATTERN = Pattern.compile("\\w+((\\w)+)?(?:,\\w+((\\w)+)?)*");
+
+    protected static final PropertyDescriptor TABLE_NAME = new PropertyDescriptor.Builder()
+            .name("Table Name")
+            .description("The name of the Kudu Table to put data into")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    static final PropertyDescriptor PREDICATES = new PropertyDescriptor.Builder()
+            .name("Predicates")
+            .description("A comma-separated list of Predicates, format: \"<colName>:<value>\" ")
+            .required(false)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.createRegexMatchingValidator(PREDICATES_PATTERN))
+            .build();
+
+    static final PropertyDescriptor PROJECTED_COLUMNS = new PropertyDescriptor.Builder()
+            .name("Projected Column Names")
+            .description("A comma-separated list of \"<column>\" names to return when scanning, default all.")
+            .required(false)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.createRegexMatchingValidator(COLUMNS_PATTERN))
+            .build();
+
+    protected static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("Batch Size")
+            .description("The maximum number of FlowFiles to process in a single execution, between 1 - 100000. " +
+                    "Depending on your memory size, and data size per row set an appropriate batch size. " +
+                    "Gradually increase this number to find out the best one for best performances.")
+            .defaultValue("500")
+            .required(false)
```

Comment:
I'd change this to true
```suggestion
            .required(true)
```